statements/downloaded_video.md

```diff
@@ -0,0 +1,64 @@
+# Downloaded a video
+
+## Description
+
+This event is emitted when a user has downloaded a file of `video` type.
+NB: this template is inspired from templates from [Video Profile](https://profiles.adlnet.gov/profile/90b2c849-d744-4d0c-8bd0-403e7859a35b).
+
+## Example
+
+```json
+{
+   "actor": {
+      "account": {
+         "name": "john",
+         "homePage": "http://gaiax.org"
+      }
+   },
+   "verb": {
+      "id": "http://id.tincanapi.com/verb/downloaded"
+   },
+   "object": {
+      "id": "http://gaiax.org/xapi/activities/ba297687-b1aa-4477-9efd-a782c8fdb90a",
+      "definition": {
+         "type": "https://w3id.org/xapi/video/activity-type/video",
+         "name": {
+            "en": "xAPI 101 Introduction"
+         }
+      }
+   },
+   "context": {
+      "contextActivities": {
+         "category": [
+            {
+               "id": "https://w3id.org/xapi/lms",
+               "definition": {
+                  "type": "http://adlnet.gov/expapi/activities/profile"
+               }
+            }
+         ]
+      }, 
+      "extensions": {
+         "https://w3id.org/xapi/video/extensions/length": 237.45,
+         "https://w3id.org/xapi/video/extensions/quality": 144,
+         "https://w3id.org/xapi/cmi5/context/extensions/sessionid": "53ff781a-3c52-11ee-be56-0242ac120002"
+      }
+   },
+   "timestamp": "2016-06-09T15:34:26.887Z"
+}
+```
+
+## Determining properties
+
+| Property | Value |
+|---|---|
+| verb.id | MUST be `http://id.tincanapi.com/verb/downloaded` |
+| object.definition.type | MUST be `https://w3id.org/xapi/video/activity-type/video` |
+
+## Rules
+
+- `context.contextActivities.category`: INCLUDED, MUST contain an activity with the `https://w3id.org/xapi/lms` id.
+- `context.extensions.https://w3id.org/xapi/video/extensions/length`: RECOMMENDED, positive float, MUST contain the value of the time length of the video file.
+- `context.extensions.https://w3id.org/xapi/video/extensions/quality`: RECOMMENDED, positive integer, MUST contain the value of the quality of the video file.
```

Comment:
What metric should be used for the quality? Is it a scale from 0 to 10? The resolution? 
 
In the [video profile](https://liveaspankaj.gitbooks.io/xapi-video-profile/content/statement_data_model.html#2626-quality), the value is a string. e.g. 360, 480, 720, 1080, 2160, sd, hd, 4k, 8k etc.

Reply:
Their specification have diverged :laughing: 
https://profiles.adlnet.gov/profile/90b2c849-d744-4d0c-8bd0-403e7859a35b/concepts/01ee0d69-3886-47c8-8828-6a55a99ccc81

Reply:
Let's stick to the official profile specification, no? 
https://github.com/adlnet/xapi-authored-profiles/blob/master/video/v1.0.3/video.jsonld#L144-L153

Reply:
Aye, another discrepancy between the video profile recipe and profile on adlnet :disappointed:
However, the [commit](https://github.com/adlnet/xapi-authored-profiles/commit/477340d38e62bfd09630462cfc55f7185ec7c2cd) that added the jsonld profile intended to align with the recipe according to the commit message.
Looking at the [recipe history](https://github.com/liveaspankaj/xapi-video-cop/commit/9c673bbf7c84ea7a571c2e1bb7943a57f85f6311), they indeed changed the quality field type from integer to string in 2017.
Also noticed that we haven't included the `quality` context extension in [ralph's video initialized statement](https://github.com/openfun/ralph/blob/master/src/ralph/models/xapi/video/contexts.py#L95-L130), nor for the video interacted statements - it's only used in the downloaded statement where it's an integer.
In [marsha](https://github.com/openfun/marsha/blob/08afc74a36e1d4677e1adea955e6345e9c9e2881/src/frontend/packages/lib_video/src/components/common/Player/videojs/xapiPlugin/index.ts#L196-L202) the quality field seems to accept both - a string and integer values, although most statements seem to contain an integer value for the quality field.